community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1928,26 +1932,58 @@ def nodeset_map(self, hostnames: list):
             nodeset_map[self.node_nodeset_name(node)].append(node)
         return nodeset_map
 
+    def _get_job_info(self, job_info, job_id=None) -> Optional[Job]:
+        """Extract job details"""
+        if job_id is None:
+            if match:= re.search(r"JobId=(\d+)", job_info):
+                job_id = match.group(1)
+
+        if match:= re.search(r"TimeLimit=(?:(\d+)-)?(\d{2}):(\d{2}):(\d{2})", job_info):
+          days, hours, minutes, seconds = match.groups()
+          duration = timedelta(
+              days=int(days) if days else 0,
+              hours=int(hours),
+              minutes=int(minutes),
+              seconds=int(seconds)
+          )
+        else:
+            duration = None
+
+        if match := re.search(r"JobName=(\w+)", job_info):
+            name = match.group(1)
+        else:
+            name = None
+
+        if match := re.search(r"JobState=(\w+)", job_info):
+            job_state = match.group(1)
+        else:
+            job_state = None
+
+        if match := re.search(r"ReqNodeList=(\w+)", job_info):
```

Comment:
Invalid regex, it'doesn't match "ReqNodeList=node-a,node-b"
Please also check if `ReqNodeList` can return "slurm hostlist" format, e.g. `node-[1-7,12,20-26]`

Reply:
Nice catch. It should gather everything up until space is detected.